src/main/java/graphql/execution/ExecutionStrategy.java

```diff
@@ -817,20 +849,20 @@ protected CompletableFuture<Object> completeValueForScalar(ExecutionContext exec
         } catch (NonNullableFieldWasNullException e) {
             return exceptionallyCompletedFuture(e);
         }
-        return completedFuture(serialized);
+        return serialized;
```

Comment:
Since this can return an error - it still can be CF but an exceptional one.

```
} catch (NonNullableFieldWasNullException e) {
            return exceptionallyCompletedFuture(e);
        }
```

Reply:
So we cant make ALL scalar / enum code ALWAYS return just an object